mne/beamformer/_lcmv.py

```diff
@@ -17,13 +17,18 @@
 from ..cov import compute_whitener, compute_covariance
 from ..source_estimate import _make_stc, SourceEstimate
 from ..source_space import label_src_vertno_sel
-from ..utils import logger, verbose, warn
+from ..utils import logger, verbose, warn, estimate_rank
 from .. import Epochs
 from ..externals import six
 
 
 def _reg_pinv(x, reg):
     """Compute a regularized pseudoinverse of a square array."""
+    # Throw a warning if cov matrix is rank-deficient but no regularization parameter is set
+    covrank = estimate_rank(x, tol='auto', norm=False, return_singular=False)
+    if(covrank < x.shape[0] and reg == 0):
```

Comment:
if(covrank < x.shape[0] and reg == 0):
->
if covrank < x.shape[0] and reg == 0.: